src/containers/PageContainer/index.tsx

```diff
@@ -1,239 +1,108 @@
-	const globalDomain = useCurrentDomain();
-	const domain = useMemo(() => location.pathname, [location.pathname]);
-
-	// Force to go back to home if invalid domain
-	useEffect(() => {
-		if (!loading && !znsDomain) {
-			return history.push(globalDomain.app);
-		}
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [znsDomain, loading, globalDomain.app]);
-
-	//- Minting State
+	const prevZnsDomain = usePrevious(znsDomain);
 	const { minted } = useMint();
 	const { fulfilled: stakingFulFilled } = useStaking();
-
-	//- Notification State
 	const { addNotification } = useNotification();
+	const { pageWidth } = usePageWidth();
+	const { modal, openModal, closeModal } = useModal();
 
-	//- Page State
-	const [hasLoaded, setHasLoaded] = useState(false);
-	const [pageWidth, setPageWidth] = useState<number>(0);
-
-	//- Overlay State
-	const [modal, setModal] = useState<Modal | undefined>();
-
-	/////////////////////
-	// Overlay Toggles //
-	/////////////////////
-
-	const openModal = useCallback(
-		(modal?: Modal) => () => setModal(modal),
-		[setModal],
-	);
-
-	const closeModal = () => {
-		setModal(undefined);
-	};
-
-	const handleResize = () => {
-		setPageWidth(window.innerWidth);
-	};
-
-	/////////////
-	// Effects //
-	/////////////
-	useEffect(() => {
-		window.addEventListener('resize', handleResize);
-		handleResize();
-		return () => {
-			window.removeEventListener('resize', handleResize);
-		};
-	}, []);
+	/**
+	 * Callback Functions
+	 */
+	const handleChainSelect = useCallback(() => {
+		if (chainId && chainSelector.selectedChain !== chainId) {
+			chainSelector.selectChain(chainId);
+		}
+	}, [chainId, chainSelector]);
 
-	/* Find the freshly minted NFT */
-	useEffect(() => {
-		if (refetch) {
-			refetch();
+	const handleForceBackHome = useCallback(() => {
+		if (!loading && !znsDomain) {
+			history.push(globalDomain.app);
 		}
-	}, [minted, refetch, stakingFulFilled]);
+	}, [loading, znsDomain, globalDomain, history]);
 
-	/* Handle notification for wallet changes */
-	useEffect(() => {
-		//wallet connect wont do this automatically if session its ended from phone
+	const handleWalletChanges = useCallback(() => {
 		if (
-			(localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) ===
-				WALLETS.WALLET_CONNECT ||
-				localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) ===
-					WALLETS.METAMASK ||
-				localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) ===
-					WALLETS.COINBASE ||
-				localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) ===
-					WALLETS.PORTIS ||
-				localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) ===
-					WALLETS.FORTMATIC) &&
+			Object.values(WALLETS).includes(
+				localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET) as WALLETS,
+			) &&
 			!active &&
 			triedEagerConnect
 		) {
-			setTimeout(async () => {
-				localStorage.removeItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET);
-			}, 2000);
+			localStorage.removeItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET);
 		}
```

Comment:
Do we gain much from using `useCallback` here vs. just having inline functions? I have a feeling that `useCallback` might be less efficient here than inline functions, but might be worth running through Profiler.